[DEV-14] feat: 단어장 페이지 좋아요 목록 필터링 기능 추가

ono212 opened the pull request:
## 📌 기능 설명
<!-- 기능을 대략적으로 설명해주세요 -->

- [x] 단어장 페이지의 좋아요 필터링 API 연동

## 📌 구현 내용
<!-- 실제 구현 내용을 디테일하게 작성해 주세요 -->

<!-- ex)
- 피그마에 디자인된 사항으로 마크업을 구현했습니다.
- 헤더 컴포넌트에 필요한 이벤트를 hook으로 구현했습니다. 
-->

원래 필터링 옵션에 '조회순'도 있어야 하는데, 아직 API에 '조회순'은 없어서 주석처리해두고 화면에도 '조회순' 옵션은 표시하지 않고 있습니다!

## 📌 논의하고 싶은 점
<!-- 논의하고 싶은 점이 있다면 적어주세요 -->
<!-- ex)
react-query 를 사용할 때 별도의 hook으로 만들어서 사용하시나요?
저는 별도로 사용하지 않는데 어떻게 하는게 좋을까요? 
-->

만약 2페이지에서 필터링 옵션을 변경하면, 필터링 옵션이 변경되고 그대로 2페이지에 있습니다. 이대로 구현하는게 나을지, 아니면 필터링 옵션을 변경했을 때 1페이지로 이동하는 것이 나은지 고민입니다! 의견주시면 감사하겠습니다🤓

https://github.com/Devminjeong-eum/frontend/assets/55550034/a2258b6d-6b55-4365-b645-625ca7ef6feb

ono212 committed:
feat: 기존에 좋아요 목록 API 연동 구현한 내역

ono212 committed:
feat: 드롭다운 정렬 옵션 상수로 생성

- 조회순은 api가 구현되지 않아서 주석 처리

ono212 committed:
feat: 드롭다운 정렬 옵션 상수로 생성

- 조회순은 api가 구현되지 않아서 주석 처리

ono212 committed:
feat: api와 hook에 정렬 param 추가

- 쿼리키 추가

ono212 committed:
feat: 드롭다운 정렬 기능 구현

- useDropdown훅으로 Dropdown 상태 분리

ono212 committed:
feat: 드롭다운 정렬 기능 구현

- useDropdown훅으로 Dropdown 상태 분리

nyoung113 commented:
만약 2페이지에서 필터링 옵션을 변경하면, 필터링 옵션이 변경되고 그대로 2페이지에 있습니다. 이대로 구현하는게 나을지, 아니면 필터링 옵션을 변경했을 때 1페이지로 이동하는 것이 나은지 고민입니다! 의견주시면 감사하겠습니다🤓
> 저는 1페이지로 이동하는게 더 맞는 것 같아요!

nyoung113 commented on src/fetcher/index.ts:
```diff
@@ -34,3 +34,24 @@ export const getWordSearch = async (wordName: string) => {
+    return await backendFetch<DefaultRes<likedWord>>(
+      `/word/like?page=${page}&limit=${limit}&sorting=${selectedOption}`,
+      {
+        headers: {
```
여기는 헤더에 어드민이 들어가있군요! 지워도 될 것 같습니답!

nyoung113 commented on src/components/pages/wordbook/index.tsx:
```diff
@@ -4,16 +4,30 @@ import { useState } from 'react';
 export default function Wordbook() {
   const [current, setCurrent] = useState(1);
-  const { data: { data: wordData, totalItems } = {} } = usePosts(current);
+  const { selectedOption, setSelectedOption } = useDropdown(
+    DROPDOWN_DEFAULT_OPTION,
+  );
+  const { data } = useGetLikedWord(
+    current,
+    10,
+    sortOptionMapping[selectedOption],
+  );
```
useDropDown에는 단순히 하나의 state를 정의하고 리턴하는 것으로 보이는데,  
이렇게 되면 useState를 직접 사용하는 것과 많이 다르지 않아서 훅을 사용하지 않아도 될 것 같다는 의견이에요! 
혹시 이 두개를 묶어서 하나의 query hook으로 만들어보는 건 어떨까요??

ono212 replied:
맞아요 3개중에 어떤 방식으로 할지 고민되더라구요ㅠㅠ 처음엔 묶어서 구현했었는데 두개를 묶어서 사용하자니 드롭다운이 재사용될 수도 있다고 생각이 들고,, 하지만 또 너무 분리했나싶고,,🤔이 부분 좀 더 고민해보겠습니다!!

nyoung113 reviewed (commented):
👍 수고하셨습니다 치치!!!👍

ono212 committed:
Merge branch 'main' into feat/DEV-14

ono212 committed:
Merge branch 'feat/DEV-14' of https://github.com/ono212/Devminjeong-eum-frontend into feat/DEV-14

ono212 committed:
refactor: 응답데이터 타입 추가, 헤더 제거

ono212 committed:
fix: 서버 응답 데이터 구조 변경에 따른 참조 수정

- data.data에서 data.data.data로 변경

ono212 committed:
feat: 필터링 옵션이 변경될 때 1페이지로 이동하도록 구현

- useDropdown에서 current상태를 관리하도록 수정
- useEffect로 current상태 변경 시 current를 1로 지정하도록 구현

ono212 commented:
> 만약 2페이지에서 필터링 옵션을 변경하면, 필터링 옵션이 변경되고 그대로 2페이지에 있습니다. 이대로 구현하는게 나을지, 아니면 필터링 옵션을 변경했을 때 1페이지로 이동하는 것이 나은지 고민입니다! 의견주시면 감사하겠습니다🤓
> 
> > 저는 1페이지로 이동하는게 더 맞는 것 같아요!

저도 그게 맞는 것 같아서 그렇게 하도록 반영해놨습니다!
